presto-hive/src/main/java/com/facebook/presto/hive/HiveUtil.java

```diff
@@ -907,6 +908,9 @@ public static String getPrefilledColumnValue(HiveColumnHandle columnHandle, Hive
             return path.toString();
         }
         if (isBucketColumnHandle(columnHandle)) {
+            if (!bucketNumber.isPresent()) {
+                throw new PrestoException(HIVE_TABLE_BUCKETING_IS_IGNORED, "Table bucketing is ignored. The virtual \"$bucket\" column cannot be referenced.");
```

Comment:
Would this also get thrown if a table isn't bucketed?

Reply:
No. If table is not bucketed - it won't have the `$bucket` column altogether, so the query will fail during the analysis.

That's actually the problem. Ideally we would like to fail such a query during the analysis. But unfortunately during the analysis we don't know yet the information about the partitions (number of partitions queried and their bucketing), thus the decision to ignore table bucketing cannot be made there.